--- gwenview-1.4.0/admin/am_edit	2006-09-25 00:04:55.213036950 +0300
+++ kdelibs-3.5.5/admin/am_edit	2006-10-03 11:40:08.052152886 +0300
@@ -499,6 +499,15 @@
         }
     }
 
+    $lookup = '(CXXLINK = \$\(LIBTOOL\).*--mode=link.*) \$\(CXXLD\)';
+    if ($MakefileData =~ m/$lookup/) {
+        substituteLine($lookup, "$1 \"\$\(CXXLD\)\"");
+    }
+    $lookup = '(LINK = \$\(LIBTOOL\).*--mode=link.*) \$\(CCLD\)';
+    if ($MakefileData =~ m/$lookup/) {
+        substituteLine($lookup, "$1 \"\$\(CCLD\)\"");
+    }
+
     $MakefileData =~ s/\$\(KCXXFLAGS\)/\$\(CXXFLAGS\)/g;
 
     $lookup = '(.*)cp -pr \$\$/\$\$file \$\(distdir\)/\$\$file(.*)';
@@ -578,6 +587,14 @@
         local @kde_options = split(/[\034\s]+/, $kde_options_str);
         if (grep(/^foreign$/, @kde_options)) {
             push(@foreignfiles, $makefileDir . "/");
+
+            # always do this. don't care if it's foreign file. we *need* CXXLD quoted for libtool argument.
+            $lookup = '(CXXLINK = \$\(LIBTOOL\).*--mode=link) \$\(CXXLD\)';
+            if ($MakefileData =~ m/$lookup/) {
+                substituteLine($lookup, "$1 \"\$\(CXXLD\)\"");
+                updateMakefile ();
+            }
+
             return 0; # don't touch me
         }
         for $opt (@kde_options) {
